File: Experiment Processing/recommender/voting_rules/average.py
from recommender.voting_rules.abstract_voting_rule import AbstractVotingRule


class Average(AbstractVotingRule):
    """
    "Average" aggregation strategy, works by calculating the average rating for each song.
    """

    def get_name(self):
        return "Average"

    def voting_rule(self, song, song_ratings):

        average = 0
        for user in song_ratings:
            average += song_ratings[user]

        return average / len(song_ratings)

File: Web Survey/resources/js/recommender/voting_rules/Average.js
import AbstractVotingRule from "./AbstractVotingRule";

export default class Average extends AbstractVotingRule {

    static getRuleName() {
        return "Average";
    }

    getDescription() {
        return "Averages all ratings for every track and picks the highest ratings"
    }

    votingRule(song, songRatings) {
        //average = 0
        let average = 0;

        //for user in song_ratings:
        //average += song_ratings[user]
        for(const user in songRatings) {
            if(songRatings.hasOwnProperty(user)) {
                average += songRatings[user];
            }
        }

        //return average / len(song_ratings)
        return average / Object.keys(songRatings).length;
    }

}